fix TestSequencer_Start

https://github.com/voedger/voedger/actions/runs/14587226010/job/40914837871
```
panic: close of closed channel
  
  goroutine 34 [running]:
  github.com/voedger/voedger/pkg/isequencer.TestSequencer_Start.func1.1()
  	/home/runner/work/voedger/voedger/pkg/isequencer/impl_test.go:84 +0x34
  github.com/voedger/voedger/pkg/isequencer.(*MockStorage).WriteValuesAndNextPLogOffset(0xc000109180, {0xb3dcc0, 0x0, 0xc000063c98?}, 0x1)
  	/home/runner/work/voedger/voedger/pkg/isequencer/test_utils.go:120 +0x184
  github.com/voedger/voedger/pkg/isequencer.(*sequencer).flusher.func1()
  	/home/runner/work/voedger/voedger/pkg/isequencer/impl.go:154 +0x94
  github.com/voedger/voedger/pkg/coreutils.Retry({0x8e1678, 0xc000014780}, {0x7f4301e57188, 0xc000014410}, 0xc000063ea8)
  	/home/runner/work/voedger/voedger/pkg/coreutils/retry.go:20 +0xe7
  github.com/voedger/voedger/pkg/isequencer.(*sequencer).flusher(0xc000166690, {0x8e1678, 0xc0000a8050})
  	/home/runner/work/voedger/voedger/pkg/isequencer/impl.go:153 +0x4fb
  created by github.com/voedger/voedger/pkg/isequencer.(*sequencer).startFlusher in goroutine 14
  	/home/runner/work/voedger/voedger/pkg/isequencer/impl.go:68 +0x1a5
```

--- a/pkg/isequencer/impl_test.go
+++ b/pkg/isequencer/impl_test.go
@@ -68,47 +68,6 @@ func TestSequencer(t *testing.T) {
 	})
 }
 
-func TestSequencer_Start(t *testing.T) {
-	require := require.New(t)
-	t.Run("should reject when too many unflushed values + allow after Flush", func(t *testing.T) {
-		iTime := coreutils.MockTime
-
-		storage := NewMockStorage()
-		params := NewDefaultParams(map[WSKind]map[SeqID]Number{
-			1: {1: 1, 2: 1, 3: 1, 4: 1, 5: 1, 6: 1},
-		})
-		params.MaxNumUnflushedValues = 5
-		waitForFlushCh := make(chan any)
-		writeOnFlushOccuredCh := make(chan any)
-		storage.onWriteValuesAndOffset = func() {
-			close(writeOnFlushOccuredCh)
-			<-waitForFlushCh
-		}
-		seq, cancel := New(params, storage, iTime)
-		defer cancel()
-
-		offset := WaitForStart(t, seq, 1, 1, true)
-		require.Equal(PLogOffset(1), offset)
-
-		// obtain 6 numbers, 6th should overflow toBeFlushed
-		for i := 0; i < 6; i++ {
-			num, err := seq.Next(SeqID(i + 1))
-			require.NoError(err)
-			require.Equal(Number(1), num)
-		}
-
-		seq.Flush()
-
-		<-writeOnFlushOccuredCh
-
-		// failed to start because toBeFlushed is overflowed
-		offset, ok := seq.Start(1, 1)
-		require.False(ok)
-		require.Zero(offset)
-		close(waitForFlushCh)
-	})
-}
-
 func TestBatcher(t *testing.T) {
 	t.Run("should aggregate max values and wait for unflushed values threshold", func(t *testing.T) {
 		require := require.New(t)

--- a/pkg/isequencer/isequencer_test.go
+++ b/pkg/isequencer/isequencer_test.go
@@ -10,6 +10,7 @@ import (
 	"math/rand"
 	"os"
 	"strconv"
+	"sync"
 	"testing"
 	"time"
 
@@ -173,6 +174,49 @@ func TestISequencer_Start(t *testing.T) {
 		},
 	})
 	params := createDefaultParams()
+	t.Run("reject on too many unflushed values + allow after Flush", func(t *testing.T) {
+		storage := isequencer.NewMockStorage()
+		params := isequencer.NewDefaultParams(map[isequencer.WSKind]map[isequencer.SeqID]isequencer.Number{
+			1: {1: 1, 2: 1, 3: 1, 4: 1, 5: 1, 6: 1},
+		})
+		params.MaxNumUnflushedValues = 5
+		waitForFlushCh := make(chan any)
+		writeOnFlushOccuredCh := make(chan any, 1)
+		once := sync.Once{}
+		storage.SetOnWriteValuesAndOffset(func() {
+			once.Do(func() {
+				close(writeOnFlushOccuredCh)
+				<-waitForFlushCh
+			})
+		})
+		seq, cancel := isequencer.New(params, storage, iTime)
+		defer cancel()
+
+		offset := isequencer.WaitForStart(t, seq, 1, 1, true)
+		require.Equal(isequencer.PLogOffset(1), offset)
+
+		// obtain 6 numbers, 6th should overflow toBeFlushed
+		for i := 0; i < 6; i++ {
+			num, err := seq.Next(isequencer.SeqID(i + 1))
+			require.NoError(err)
+			require.Equal(isequencer.Number(1), num)
+		}
+
+		seq.Flush()
+
+		<-writeOnFlushOccuredCh
+
+		// failed to start because toBeFlushed is overflowed
+		offset, ok := seq.Start(1, 1)
+		require.False(ok)
+		require.Zero(offset)
+		close(waitForFlushCh)
+
+		// now ok to start because the queue is flushed
+		// will wait for start because the queue will be flushed not immediately
+		offset = isequencer.WaitForStart(t, seq, 1, 1, true)
+		require.Equal(isequencer.PLogOffset(2), offset)
+	})
 
 	t.Run("should panic when cleanup process is initiated", func(t *testing.T) {
 		sequencer, cleanup := isequencer.New(params, storage, iTime)

--- a/pkg/isequencer/isequencer_test.go
+++ b/pkg/isequencer/isequencer_test.go
@@ -181,11 +181,11 @@ func TestISequencer_Start(t *testing.T) {
 		})
 		params.MaxNumUnflushedValues = 5
 		waitForFlushCh := make(chan any)
-		writeOnFlushOccuredCh := make(chan any, 1)
+		writeOnFlushOccurredCh := make(chan any, 1)
 		once := sync.Once{}
 		storage.SetOnWriteValuesAndOffset(func() {
 			once.Do(func() {
-				close(writeOnFlushOccuredCh)
+				close(writeOnFlushOccurredCh)
 				<-waitForFlushCh
 			})
 		})

--- a/pkg/isequencer/test_utils.go
+++ b/pkg/isequencer/test_utils.go
@@ -69,6 +69,12 @@ func (m *MockStorage) SetReadNextPLogOffsetError(err error) {
 	m.readNextOffsetError = err
 }
 
+func (m *MockStorage) SetOnWriteValuesAndOffset(f func()) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	m.onWriteValuesAndOffset = f
+}
+
 func (m *MockStorage) SetOnActualizeFromPLog(f func()) {
 	m.mu.Lock()
 	defer m.mu.Unlock()

